src/Service.Host/client/src/components/Pallets.js

```diff
@@ -0,0 +1,596 @@
+                            </Grid>
+                        </Grid>
+                    </Grid>
+                    <Grid size={12}>
+                        <Grid container spacing={2}>
+                            <Grid item size={2}>
+                                <Dropdown
+                                    value={pallet.auditedByEmployee}
+                                    fullWidth
+                                    propertyName="lastAuditedBy"
+                                    label="Last Audited By"
+                                    allowNoValue
+                                    items={employeeNames}
+                                    onChange={handleEmployeeFieldChange}
+                                />
+                            </Grid>
+
+                            <Grid item size={2}>
+                                <DatePicker
+                                    label="Date Last Audited"
+                                    propertyName="dateLastAudited"
+                                    value={pallet?.dateLastAudited}
+                                    onChange={date => handleDateChange('dateLastAudited', date)}
+                                    clearable
+                                />
+                            </Grid>
+                            <Grid item size={2}>
+                                <InputField
+                                    propertyName="auditFrequencyWeeks"
+                                    label="Audit Frequency Weeks"
+                                    type="number"
+                                    onChange={handleFieldChange}
+                                    value={pallet?.auditFrequencyWeeks}
+                                    fullWidth
+                                />
+                            </Grid>
+                            <Grid item size={2}>
+                                <InputField
+                                    propertyName="auditedByDepartmentCode"
+                                    label="Audited By Department Code"
+                                    value={pallet?.auditedByDepartmentCode}
+                                    disabled
+                                    fullWidth
+                                />
+                            </Grid>
+                            <Grid item size={4}>
+                                <Search
+                                    autoFocus
+                                    propertyName="auditedByDepartment"
+                                    label="Audited By Department"
+                                    resultsInModal
+                                    resultLimit={100}
+                                    value={pallet?.auditedByDepartmentDescription}
+                                    loading={departmentSearchLoading}
+                                    handleValueChange={(_, newVal) =>
+                                        setPallet({
```

Comment:
again, use the functional variation of setPallet here, so setPallet(p => ({...p, etc}))